contracts/governance/Staking/Staking.sol

```diff
@@ -323,6 +366,24 @@ contract Staking is IStaking, WeightedStaking, ApprovalReceiver {
 		emit TokensWithdrawn(msg.sender, receiver, amount);
 	}
 
+	// @dev withdraws tokens for lock date 2 weeks later than given lock date
+	function _withdrawNext(
+		uint96 amount,
+		uint256 until,
+		address receiver,
+		bool isGovernance
+	) internal {
+		if (msg.sender.isContract()) {
```

Comment:
is it an attempt to identify vesting contract? please be more specific

Reply:
added function _isVestingContract